Add QuicTransportOptions configuration documentation to HTTP/3 guide (#36341)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/servers/kestrel/http3.md

For more information, see <xref:fundamentals/servers/kestrel/endpoints>.

## Configure QuicTransportOptions

> QUIC transport options can be configured by calling the <xref:Microsoft.AspNetCore.Hosting.WebHostBuilderQuicExtensions.UseQuic%2A> extension method on <xref:Microsoft.AspNetCore.Hosting.IWebHostBuilder>. The `UseQuic` method accepts an `Action<QuicTransportOptions>` delegate that configures the QUIC transport.

Reviewer: @copilot 

```suggestion
QUIC transport options can be configured by calling the <xref:Microsoft.AspNetCore.Hosting.WebHostBuilderQuicExtensions.UseQuic%2A> extension method on <xref:Microsoft.AspNetCore.Hosting.IWebHostBuilder>.
```

> :::code language="csharp" source="samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_UseQuicWithOptions" highlight="3-5":::

Reviewer: ```suggestion
:::code language="csharp" source="samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_UseQuicWithOptions" highlight="3-8":::
```

> :::code language="csharp" source="samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_UseQuicWithOptions" highlight="4-6":::

Reviewer: ```suggestion
:::code language="csharp" source="samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_UseQuicWithOptions" highlight="3-8":::
```

> The following table describes the available <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportOptions>:

Reviewer 2: @copilot Do not end a sentence before a table with a colon. Always end a sentence before a table with a period.

|--------|---------|-------------|
| <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportOptions.MaxBidirectionalStreamCount> | `100` | The maximum number of concurrent bi-directional streams per connection. |
| <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportOptions.MaxUnidirectionalStreamCount> | `10` | The maximum number of concurrent inbound uni-directional streams per connection. |
| <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportOptions.MaxReadBufferSize> | `1024 * 1024` (1 MB) | The maximum read buffer size in bytes. |
| <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportOptions.MaxWriteBufferSize> | `64 * 1024` (64 KB) | The maximum write buffer size in bytes. |
| <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportOptions.Backlog> | `512` | The maximum length of the pending connection queue. |
| <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportOptions.DefaultStreamErrorCode> | `0` | Error code used when the stream needs to abort the read or write side of the stream internally. The value must be between `0x0` and `0x3FFFFFFFFFFFFFFF`. |

> | <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportOptions.DefaultCloseErrorCode> | `0` | Error code used when an open connection is disposed. The value must be between `0x0` and `0x3FFFFFFFFFFFFFFF`. |

Reviewer: @copilot 

```suggestion
| <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportOptions.DefaultStreamErrorCode> | `0x010c` (`H3_REQUEST_CANCELLED`) | Error code used when the stream needs to abort the read or write side of the stream internally. The value must be between `0x0` and `0x3FFFFFFFFFFFFFFF`. |
| <xref:Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportOptions.DefaultCloseErrorCode> | `0x100` (`H3_NO_ERROR`) | Error code used when an open connection is disposed. The value must be between `0x0` and `0x3FFFFFFFFFFFFFFF`. |
```
